js/main.js

```diff
@@ -1,103 +1,31 @@
-import $ from 'jquery';
-import Mustache from 'mustache';
-import {
-  reportCount,
-  reportError,
-  versionString,
-  settingsError
-} from './consort-template';
-import VERSION_STRING from './version';
+import 'babel-polyfill';
 
-const consortReportContainer = '.consort-report';
-const consortReportVersionContainer = '.consort-report-version';
+import Vue from 'vue';
 
-/**
- * Render consort report for defined reports. Because reports are expensive, this function recursively retrieves and
- * renders each report one at a time. Errors are reported but do not prevent the next report from being processed.
- *
- * @param {Object} config - Module configuration
- * @param {String} dataUrl - The external modules URL for `lib/data.php`.
- * @return {Integer} order - A variable for keeping track of survey order based on the config.reports order. Should
- *   be treated as private - only set internally during recursion.
- */
-export function buildConsortReport(config, dataUrl, order=0) {
-  const total = config.length;
-  let next = order + 1;
-  if (order === total) {
-    return;
-  }
-  const report = config[order];
-  $.ajax({
-    url: dataUrl,
-    data: {
-      "reportId": report.reportId
-    },
-    method: "POST",
-    success: function(response) {
-      $(consortReportContainer).append(reportHeading(report.name, response.count));
-      buildConsortReport(config, dataUrl, next);
-    },
-    error: function(response) {
-      $(consortReportContainer).append(errorMessage(report.name));
-      buildConsortReport(config, dataUrl, next);
-    }
-  });
-}
-
-/**
- * A function to produce HTML for a report heading.
- * @param {String} reportName - Name of report
- * @param {Integer} totalRecords - Total number of records in report.
- * @return {String} HTML for report heading
- */
-export function reportHeading(reportName, totalRecords) {
-  return Mustache.render(reportCount, { name: reportName, count: totalRecords });
-}
-
-/**
- * A function to produce HTML for error messages when report counts can not be retrieved.
- * @param {String} reportName - Name of report
- * @return {String} HTML for report error message.
- */
-export function errorMessage(reportName) {
-  return Mustache.render(reportError, { name: reportName });
-}
+import ConsortReport from './components/ConsortReport';
 
 /**
  * Main method that will render a consort diagram.
  */
 export function run(urlsJsonString) {
   const urls = JSON.parse(urlsJsonString);
-  $.ajax({
-    url: urls.settings,
-    method: "POST",
-    success: function(config) {
-      buildConsortReport(config, urls.data);
+  new Vue({
+    el: '.consort-report-container',
+    components: { ConsortReport },
+    render(createElement) {
+      return createElement(ConsortReport, {
+        props: {
+          pid: this.pid
+        }
+      });
+    },
+    data: {
+      pid
```

Comment:
Where is `pid` defined? It also looks like you're not declaring a prop for it in the `ConsortReport` component.

Reply:
I left this in here thinking I would need it for the subsequent work but I don't think it was needed. I removed it and the `urls` constant for now.